src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -49,21 +53,23 @@ public bool HasSpanThatContains(SnapshotPoint point)
         }
 
         public IList<ITagSpan<TTag>> GetIntersectingSpans(SnapshotSpan snapshotSpan)
+        {
+            var pooledResult = Classifier.GetPooledList<ITagSpan<TTag>>(out var result);
+            AddIntersectingSpans(snapshotSpan, result);
+            return Classifier.GetFinalList(pooledResult);
+        }
+
+        private void AddIntersectingSpans(SnapshotSpan snapshotSpan, SegmentedList<ITagSpan<TTag>> result)
         {
             var snapshot = snapshotSpan.Snapshot;
             Debug.Assert(snapshot.TextBuffer == _textBuffer);
 
-            var intersectingIntervals = _tree.GetIntervalsThatIntersectWith(
-                snapshotSpan.Start, snapshotSpan.Length, new IntervalIntrospector(snapshot));
+            using var intersectingIntervals = TemporaryArray<TagNode>.Empty;
```

Comment:
>TemporaryArray [](http://example.com/codeflow?start=46&length=14)

This is best when < 5 entries, right? Is that the common case for this codepath?

Reply:
yup.  it's a common path.  i instrumented this one time and it was something like 80% less than that, and 95% less than 8.  i would like to acutally increase the size of hte buffer here as it would make a lot more cases stop hitting the heap.